sdks/python/apache_beam/transforms/core.py

```diff
@@ -1712,6 +1718,10 @@ def __eq__(self, other):
           and self.timestamp_combiner == other.timestamp_combiner)
     return False
 
+  def __hash__(self):
+    return hash((type(self), self.windowfn, self.accumulation_mode,
```

Comment:
Unfortunately removing `__hash__` here, results in `TypeError: unhashable type: 'Windowing'` for ` test_top_prefixes`. 

However, looking at self._default, it seems OK to remove it from the hash, since it's actually a check on the other class variables. Therefore I think `__eq__` and `__hash__` are still in sync. I removed `type(self)` as well.

Reply:
You are right, `self._default` indeed is defined based on values of other object attributes included in hash. 

Objects of this class are used in this dictionary: https://github.com/apache/beam/blob/b8499d9ce9d00bb78049abedc929a8d5d59fdfbd/sdks/python/apache_beam/runners/pipeline_context.py#L78. Current implementation makes sense to me.